File: api/skill.js
// Vercel 환경에서 body를 수동으로 파싱
export default function handler(req, res) {
  console.log('Request received:', req.method);
  
  if (req.method === 'GET') {
    return res.status(200).json({ status: 'OK' });
  }

  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  // 기본 응답 (일단 하드코딩으로 테스트)
  const response = {
    "version": "2.0",
    "template": {
      "outputs": [{
        "simpleText": {
          "text": "안녕하세요! 카카오 스킬이 정상 작동합니다."
        }
      }]
    }
  };

  console.log('Sending basic response');
  res.setHeader('Content-Type', 'application/json; charset=utf-8');
  return res.status(200).json(response);
}

// body 파싱을 위한 설정
export const config = {
  api: {
    bodyParser: {
      sizeLimit: '1mb',
    },
  },
}

File: api/skill.py
from http.server import BaseHTTPRequestHandler
import json

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        self.send_response(200)
        self.send_header('Content-Type', 'application/json')
        self.end_headers()
        response = {"status": "OK", "message": "Skill server is running"}
        self.wfile.write(json.dumps(response).encode())

    def do_POST(self):
        try:
            # 요청 본문 읽기
            content_length = int(self.headers.get('Content-Length', 0))
            if content_length > 0:
                post_data = self.rfile.read(content_length)
                body = json.loads(post_data.decode('utf-8'))
                user_text = body.get("userRequest", {}).get("utterance", "")
            else:
                user_text = ""

            # 응답 생성
            response_data = {
                "version": "2.0",
                "template": {
                    "outputs": [
                        {
                            "simpleText": {
                                "text": f"안녕하세요! '{user_text}' 라고 하셨네요."
                            }
                        }
                    ]
                }
            }

            # 응답 전송
            self.send_response(200)
            self.send_header('Content-Type', 'application/json; charset=utf-8')
            self.end_headers()
            self.wfile.write(json.dumps(response_data, ensure_ascii=False).encode('utf-8'))

        except Exception as e:
            print(f"Error: {e}")
            self.send_response(500)
            self.send_header('Content-Type', 'application/json; charset=utf-8')
            self.end_headers()
            error_response = {
                "version": "2.0",
                "template": {
                    "outputs": [
                        {
                            "simpleText": {
                                "text": "서버 오류가 발생했습니다."
                            }
                        }
                    ]
                }
            }
            self.wfile.write(json.dumps(error_response, ensure_ascii=False).encode('utf-8'))